openssl/src/ssl/mod.rs

```diff
@@ -1120,6 +1120,50 @@ impl SslContextBuilder {
         }
     }
 
+    /// Sets the callback for generating a cookie for stateless handshakes.
+    ///
+    /// The callback will be called with the SSL context and a slice into which the cookie
+    /// should be written. The callback should return the number of bytes written.
+    ///
+    /// This corresponds to `SSL_CTX_set_cookie_generate_cb`.
+    #[cfg(all(feature = "v111", ossl111))]
+    pub fn set_cookie_generate_cb<F>(&mut self, callback: F)
+    where
+        F: Fn(&mut SslRef, &mut [u8]) -> Result<usize, ()> + 'static + Sync + Send
```

Comment:
Callbacks normally return an `ErrorStack` on error.

Reply:
How should I come up with one, and what good would it do? The OpenSSL implementation only checks whether the return code is nonzero.

Reply:
Take a look at the other callbacks - right now we just throw the error away but the intent is to push the state back onto the thread local error stack.

Reply:
The other callbacks punt the problem to the user. The user isn't expected to be making any SSL-related calls inside those two callbacks, so there's nowhere for an SSL error to come from.

Reply:
What problem is being punted?

Reply:
Coming up with some meaningful openssl::Error value.

Reply:
There is only one way to construct an `ErrorStack`.

Reply:
I'm confused. Why is application code supposed to generate a SSL error?

Reply:
In other callbacks it would be produced by some other openssl function. For example, the `tmp_dh` callback may hit an error from `Dh::params_from_pem`. The potential source here could be `rand::rand_bytes` for example.

In any case, it is strictly more powerful to return `ErrorStack` than `()`, and matches with every single other callback function.

Reply:
That strikes me as a very strange idiom, especially as it is my current understanding that implementations of this callback will typically be pure and total MAC computations on a source address. Still, if you insist, it's not like it has any more serious drawbacks than weirdness.

Reply:
If these callbacks can never fail, it should just return `usize`, yes?